packages/backend/src/spsp/service.ts

```diff
@@ -63,9 +63,13 @@ async function getPay(
     )
   }
 
-  const user = await deps.userService.get(ctx.params.id)
-  const account = user && (await deps.accountService.get(user.accountId))
-  if (!account) {
+  const accountId = ctx.params.id
+  const invoice = await deps.wmService
+    .getCurrentInvoice(accountId)
+    .catch(() => {
```

Comment:
Why not `500` on errors? (eg. a database error, not a missing record)

Reply:
So the problem I'm trying to solve here is that if the underlying account doesn't exist to return the 404 error that was previously defined
```
ctx.status = 404
    ctx.set('Content-Type', CONTENT_TYPE_V4)
    ctx.body = JSON.stringify({
      id: 'InvalidReceiverError',
      message: 'Invalid receiver ID'
    })
```

Obviously the above is not great as it swallows all errors. I can either leave it like this or create a specific error for when an account can't be found or something else? What do you recommend @sentientwaffle

Reply:
The account is fetched later anyway (in the `try..catch`). If that was moved up, and `accountService.get` returned `Account | undefined` (instead of `throw`-ing on an absent account) it could trigger a `404`.

Reply:
Ok, I have moved things around to better solve that case now